src/main/java/baritone/utils/schematic/format/defaults/StructureNBT.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.utils.schematic.format.defaults;
+
+import baritone.utils.accessor.ITemplate;
+import baritone.utils.schematic.StaticSchematic;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.gen.structure.template.Template;
+
+import java.util.List;
+
+/**
+ * @author Sam
+ * @since 03/11/2022
+ */
+public final class StructureNBT extends StaticSchematic {
+
+    public StructureNBT(NBTTagCompound schematic) {
+        Template template = new Template();
+        template.read(schematic);
+
+        BlockPos size = template.getSize();
+        this.x = size.getX();
+        this.y = size.getY();
+        this.z = size.getZ();
+
+        List<Template.BlockInfo> blocks = ((ITemplate) template).getBlocks();
+        this.states = new IBlockState[this.x][this.z][this.y];
+        for (Template.BlockInfo block : blocks ) {
+            this.states[block.pos.getX()][block.pos.getZ()][block.pos.getY()] = block.blockState;
+        }
+    }
+
+    @Override
+    public boolean inSchematic(int x, int y, int z, IBlockState currentState) {
+        // Filtering out Structure Void
+        if (x >= 0 && x < widthX() && y >= 0 && y < heightY() && z >= 0 && z < lengthZ()) {
+            return this.states[x][z][y] != null;
+        }
+        return false;
+    }
```

Comment:
If this check actually filters out null states, can you maybe try using a schematic with null states with `mapArtMode` enabled?

Reply:
What would the right aproach to fix this be? We want to prevent any `null` of leaking through `getDirect` and `getColumn`, but there would have to be some sort of placeholder (because otherwise there would be a hight shift in the column).

The easiest solution i can think of is to fill `this.states` with Air blocks instead of null while still keeping track of the structure void. That way if the blocks get read without checking `inSchematic` there would not be a NullPointerException, but the blocks would hopefully get ignored when building.

Reply:
Ok, so i fixed this by overriding `getDirect`, `getColumn` to use air as placeholders and `desiredState` to return the current block.
There is still `null` in `this.states`, but since that can only be accessed from within subclasses (as far as i understand) it shouldn't be a problem.

Reply:
It's not tedious.
My current fight against the gradle cache is tedious.